third_party/infineon/cyw30739_sdk/scripts/gen_ss_cgs.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""SS CGS generator
+
+This script generates SS CGS and HDF files.
+
+"""
+
+import argparse
+import io
+import os
+import pathlib
+import sys
+
+from collections import OrderedDict
+
+THREAD_FACTORY_KEY_BASE = 0x2000
+MATTER_FACTORY_KEY_BASE = 0x2100
+
+
+def main():
+    args = parse_args()
+
+    configs = OrderedDict()
+    add_thread_factory_config(configs)
+
+    gen_ss_hdf(args.ss_hdf, configs)
+    gen_ss_cgs(args.ss_cgs, args.ss_hdf, configs)
+    gen_empty_cgs(args.empty_cgs, args.ss_hdf, configs)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--ss_hdf", required=True, type=pathlib.Path)
+    parser.add_argument("--ss_cgs", required=True, type=pathlib.Path)
+    parser.add_argument("--empty_cgs", required=True, type=pathlib.Path)
+    return parser.parse_args()
+
+
+def add_thread_factory_config(configs: OrderedDict):
```

Comment:
why is the param in/out?

usually it is more pythonic to return, so you would have `def get_thread_factory_config() -> OrderedDict`

Reply:
fixed